modules/ximgproc/include/opencv2/ximgproc/edge_drawing.hpp

```diff
@@ -0,0 +1,81 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef __OPENCV_EDGE_DRAWING_HPP__
+#define __OPENCV_EDGE_DRAWING_HPP__
+
+#include <opencv2/core.hpp>
+
+namespace cv
+{
+namespace ximgproc
+{
+
+//! @addtogroup ximgproc_fast_line_detector
+//! @{
+
+class CV_EXPORTS_W EdgeDrawing : public Algorithm
+{
+public:
+
+    enum GradientOperator
+    {
+        PREWITT = 0,
+        SOBEL = 1,
+        SCHARR = 2,
+        LSD = 3
+    };
+
+    enum ImageStyle
+    {
+        DRAW_NONE = 0,
+        DRAW_CIRCLES = 1,
+        DRAW_ELLIPSES = 2,
+        DRAW_BOTH = 3
+    };
+
+    enum EllipseFittingMethods
+    {
+        BOOKSTEIN = 0,
+        FPF = 1
+    };
+
+    CV_WRAP virtual void setInput(InputArray image) = 0;
+    CV_WRAP virtual void getEdgeImage(OutputArray dst) = 0;
+    CV_WRAP virtual void getGradImage(OutputArray dst) = 0;
+
+    CV_WRAP virtual std::vector<std::vector<Point> > getSegments() = 0;
+
+    CV_WRAP virtual void getLines(InputArray src, OutputArray lines, bool validate = true) = 0;
+    CV_WRAP virtual void detectEllipses(OutputArray ellipses, bool validate = true) = 0;
```

Comment:
> EdgeDrawing
> getLines
> detectEllipses

Is it an edge detector?
Where from `EdgeDrawing` name come?

Reply:
"EdgeDrawing" the name explained in [The paper](http://c-viz.eskisehir.edu.tr/pdfs/ED.pdf)
as:
" Unlike traditional edge detectors, which work on the
thresholded gradient magnitude cluster to determine edge elements, our method first spots sparse points
along rows and columns called anchors, and then joins these anchors via a smart, heuristic edge tracing
procedure, hence the name Edge Drawing (ED). ED produces edge maps that always consist of clean, perfectly contiguous, well-localized, one-pixel wide edges"